test/LPIncentiveHook.t.sol

```diff
@@ -1558,15 +1543,109 @@ contract LPIncentiveHookTest is Test, Deployers {
+        vm.startPrank(alice);
+        IERC20(Currency.unwrap(token0)).approve(address(modifyLiquidityRouter), type(uint256).max);
+        IERC20(Currency.unwrap(token1)).approve(address(modifyLiquidityRouter), type(uint256).max);
+        vm.stopPrank();
+
+        // Create test params
+        int24 tickLower = -120;
+        int24 tickUpper = 120;
+        uint256 liquidity = 100e18;
+
+        // Add liquidity with alice as the reward target
+        vm.prank(alice);
+        modifyLiquidityRouter.modifyLiquidity(
+            key,
+            IPoolManager.ModifyLiquidityParams({
+                tickLower: tickLower,
+                tickUpper: tickUpper,
+                liquidityDelta: int256(liquidity),
+                salt: bytes32(0)
+            }),
+            abi.encode(alice)
+        );
+
+        // Trying to add more liquidity to the same position but with bob as reward target should fail
+        vm.prank(alice);
+        // vm.expectRevert(LPIncentiveHook.PositionAlreadyHasRewardTarget.selector); \\todo: make sure we test the right error
+        vm.expectRevert();
+        modifyLiquidityRouter.modifyLiquidity(
+            key,
+            IPoolManager.ModifyLiquidityParams({
+                tickLower: tickLower,
+                tickUpper: tickUpper,
+                liquidityDelta: int256(liquidity),
+                salt: bytes32(0)
+            }),
+            abi.encode(bob)
+        );
+    }
+
+    function test_CanReuseRewardTarget() public {
+        // Deal tokens to alice
+        deal(Currency.unwrap(token0), alice, 100000 ether);
+        deal(Currency.unwrap(token1), alice, 100000 ether);
+
+        // Approve tokens for router
+        vm.startPrank(alice);
+        IERC20(Currency.unwrap(token0)).approve(address(modifyLiquidityRouter), type(uint256).max);
+        IERC20(Currency.unwrap(token1)).approve(address(modifyLiquidityRouter), type(uint256).max);
+        vm.stopPrank();
+
+        // Create test params
+        int24 tickLower = -120;
+        int24 tickUpper = 120;
+        uint256 liquidity = 1000e18;
+
+        // Add liquidity with alice as the reward target
+        vm.prank(alice);
+        modifyLiquidityRouter.modifyLiquidity(
+            key,
+            IPoolManager.ModifyLiquidityParams({
+                tickLower: tickLower,
+                tickUpper: tickUpper,
+                liquidityDelta: int256(liquidity),
+                salt: bytes32(0)
+            }),
+            abi.encode(alice)
+        );
+
+        // Adding more liquidity to the same position with the same reward target should succeed
+        vm.prank(alice);
+        modifyLiquidityRouter.modifyLiquidity(
+            key,
+            IPoolManager.ModifyLiquidityParams({
+                tickLower: tickLower,
+                tickUpper: tickUpper,
+                liquidityDelta: int256(liquidity),
+                salt: bytes32(0)
+            }),
+            abi.encode(alice)
```

Comment:
maybe here the `addLiquidity()` function could be used too? or `adjustLiquidity`

Reply:
done